org.eclipse.jdt.core.javac/src/org/eclipse/jdt/internal/javac/JavacProblemConverter.java

```diff
@@ -997,6 +997,17 @@ yield switch (rootCauseCode) {
 			case "compiler.err.switch.expression.empty" -> IProblem.SwitchExpressionsYieldMissingDefaultCase;
 			case "compiler.err.return.outside.switch.expression" -> IProblem.SwitchExpressionsReturnWithinSwitchExpression;
 			case "compiler.err.cant.apply.diamond.1" -> IProblem.NonGenericType;
+			case "compiler.err.class.in.unnamed.module.cant.extend.sealed.in.diff.package" -> IProblem.SealedPermittedTypeOutsideOfPackage;
+			case "compiler.err.non.sealed.or.sealed.expected" -> IProblem.SealedMissingInterfaceModifier;
+			case "compiler.err.array.dimension.missing" -> IProblem.MustDefineEitherDimensionExpressionsOrInitializer;
+			case "compiler.warn.deprecated.annotation.has.no.effect" -> 0; // ignore this warning
+			case "compiler.err.enum.constant.not.expected" -> IProblem.UndefinedMethod;
+			case "compiler.warn.poor.choice.for.module.name" -> IProblem.ModuleRelated;
+			case "compiler.err.try.without.catch.finally.or.resource.decls" -> IProblem.Syntax;
+			case "compiler.warn.unchecked.meth.invocation.applied" -> IProblem.UnsafeTypeConversion;
+			case "compiler.err.encl.class.required" -> IProblem.MissingEnclosingInstanceForConstructorCall;
+			case "compiler.err.operator.cant.be.applied" -> IProblem.InvalidOperator;
+			case "compiler.warn.try.resource.not.referenced" -> 0; // ECJ doesn't do this
```

Comment:
This came into discussion today. Should we "ignore" like done here or we should find a way to report such warnings as this is clear benefit from using javac? See https://github.com/eclipse-jdtls/eclipse-jdt-core-incubator/issues/795

Reply:
I'll take Fred's suggestion of finding the closest category of error and using that as the problem id. As well, I'll comment that these are not in ECJ.